### DISCUSSION TITLE
Docs: -store-gateway.sharding-ring.heartbeat-timeout

### DISCUSSION THREAD

**Discussion by andreivaloshyn** [grafana/mimir#3025] [Help and support] [answered]
#### Is your documentation request related to a feature? If so, which one?

Documentation for `-store-gateway.sharding-ring.heartbeat-timeout` told us that 
```
  # This option needs be set both on the store-gateway, querier and ruler when running in
  # microservices mode.
 ``` 
However querier does not have this option documented

#### Describe the solution that you’d like or the expected outcome

Update documentation

**Comment by pracucci** [comment, marked as answer]
👋 Mimir has several configuration options that are shared between multiple components. For example, the `-store-gateway.sharding-ring.heartbeat-timeout` CLI flag (or its respective YAML config option) must be set to store-gateway, querier and ruler when deployed in microservices mode. The querier and ruler config name is not different, they just accept `-store-gateway.sharding-ring.heartbeat-timeout` too.

Does this clarify it?

**Comment by andreivaloshyn** [comment]
Hey @pracucci and thanks for your response
I added 
```
      sharding_ring:
        heartbeat_timeout: 2m
```
to the querier's and ruler's yaml sections
And got this error
```
error loading config from /etc/mimir/mimir.yaml: Error parsing config file: yaml: unmarshal errors:
  line 119: field sharding_ring not found in type querier.Config
  line 128: field sharding_ring not found in type ruler.Config
```
Does it mean I did something wrong?

**Comment by 56quarters** [comment]
The setting should only be added to the `store_gateway` section. The ruler and querier will read the `store_gateway` section to get the value. You just need make sure the ruler and querier are reading the same configuration file or configmap. If you are using CLI flags, the ruler and querier will need the `-store-gateway.sharding-ring.heartbeat-timeout` flag as well like @pracucci mentioned.

**Reply by andreivaloshyn** [reply]
Thank you @56quarters @pracucci 
Seems it's working now.
